### PR TITLE
160957934 improve malformed interface url detection

### PR DESCRIPTION
**Pull request by K1ll3rF0x**
# Added
* Zip package validation when unzipping.
* Zip file content listing without extracting utility function.
* Zip file list validation in transport service form.

# Changed
* Improved GTFS and Kalkati.net interface zip package validation. Checking that required files must be present.

### CODE REVIEW

File: ote/src/clj/ote/integration/import/gtfs.clj [hunk 1]
```diff
@@ -220,13 +216,63 @@
...
+
+(defn check-interface-zip [type db interface-id url byte-array-data]
+  (try
+    (validate-interface-zip-package type byte-array-data)
+
+    (catch Exception e
+      (log/warn "Error when opening interface zip package from url" url ":" (.getMessage e))
+      (specql/update! db ::t-service/external-interface-description
+                      {::t-service/gtfs-imported (java.sql.Timestamp. (System/currentTimeMillis))
+                       ::t-service/gtfs-import-error (str "Invalid interface package: " (.getMessage e))}
```
Comment (rckopteri) [inline ote/src/clj/ote/integration/import/gtfs.clj]: gtfs-import-error is shown in ui. I think that it should be in finnish or even better, it should be translated.

File: ote/src/cljs/ote/views/transport_service_common.cljs [hunk 2]
```diff
@@ -170,7 +170,7 @@
                        :tooltip (tr [:form-help :external-interfaces-tooltips :external-service-url])
                        :width "20%"
                        :full-width? true
-                       :on-blur #(e! (ts/->EnsureExternalInterfaceUrl (-> % .-target .-value)))
+                       :on-blur #(e! (ts/->EnsureExternalInterfaceUrl (-> % .-target .-value) "GTFS")) ;; TODO: Get format from state
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_service_common.cljs]: Is this hardcoded gtfs value ok for now?

Comment (K1ll3rF0x) [reply ote/src/cljs/ote/views/transport_service_common.cljs]: Nope, still working on it.

File: ote/src/cljs/ote/views/transport_service_common.cljs [hunk 3]
```diff
@@ -165,32 +166,55 @@
...
+
+                                     [:span {:style {:width "30%" :margin-left "10px"}}
+                                      (let [url-status (get-in external-interface [:url-status :status])
+                                            url-error (get-in external-interface [:url-status :error])]
+                                        (if-not url-status
+                                          [gtfs-viewer-link service]
+
+                                          (if (= :success url-status)
+                                            [:span {:style {:display "flex" :flex-flow "row nowrap"}}
+                                             [(tooltip-wrapper ic/action-done) {:style (merge style-base/icon-small
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_service_common.cljs]: Consider moving style changes to stylefy completely.

File: ote/src/cljs/ote/views/transport_service_common.cljs [hunk 4]
```diff
@@ -165,32 +166,55 @@
...
+                                                                                              {:color "green"
+                                                                                               :position "relative"
+                                                                                               :top "15px"})}
+                                              {:text (tr [:field-labels :transport-service-common :external-interfaces-ok])}]
+                                             [gtfs-viewer-link service]]
+                                            [:span [(tooltip-wrapper ic/alert-warning) {:style (merge style-base/icon-small
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_service_common.cljs]: use stylefy

**Pull request merged**